lib/postgrex/protocol.ex

```diff
@@ -332,11 +339,12 @@ defmodule Postgrex.Protocol do
   defp init_recv(%{timeout: timeout} = s, status, buffer) do
     case msg_recv(s, timeout, buffer) do
       {:ok, msg_backend_key(pid: pid, key: key), buffer} ->
-        init_recv(%{s | backend_key: {pid, key}}, status, buffer)
+        init_recv(%{s | connection_id: pid, backend_key: {pid, key}}, status, buffer)
```

Comment:
Can we not duplicate this? Since we aren't using the `backend_key` we could remove it, at least until we need it.